crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -270,4 +424,553 @@ with UI components as a child of an entity without UI components, results may be
             Err(LayoutError::InvalidHierarchy)
         }
     }
+
+    /// Returns a debug representation of the computed layout of the UI layout tree for each camera.
+    pub fn ui_layout_tree_debug_string(&self) -> Result<String, fmt::Error> {
```

Comment:
This feels weird, as noted in the PR description. Open to alternatives. Maybe the `pub(super)` get accessors become available outside of test and debug defines a trait that the user can manually import?